### PR TITLE
generate openshift-template for operator installation

### PR DESCRIPTION
**Pull request by geoberle**
`hypershift install --render` provides a way to generate the installation manifests
for the hypershift operator in a multi-doc yaml format. in order to support more
installation options, this change adds the capability to generate an openshift-template.

to make room for more flags influencing the rendering process, a new subcommand
has been introduced and the existing rendering code has been moved accordingly
while still leveraging the core manifest generation function `hyperShiftOperatorManifests`.
all the flags from the `install` command have been marked as `PersistentFlags` so
they are available in the `render` subcommand. the previously supported `--render` flag
has been removed.

to accommodate the need to inject template parameters into the OIDC secret,
`HyperShiftOperatorOIDCProviderS3Secret` has been changed to generate
`stringData` instead of `data`.

to inject template parameter placeholders to fields that are not strings, the
strictly typed runtime objects generated by `hyperShiftOperatorManifests`
are converted into `unstructured.Unstructured` during template generation.
this does not affect the regular `install` code path which still operates on
the regular runtime objects.

the app-sre-saas-template Makefile target has been changed to use the `--template`
flag. this target is now also part of the verify target to prevent latent template
changes not being commited along with code changes.

json list rendering groups all manifests under one `v1/List` manifest while
yaml list rendering still leverages the `---` manifest delimiter.

```
hypershift install <flags> render [--format yaml|json]

hypershift install <flags> render --template [--format yaml|json]
```

Part of https://issues.redhat.com/browse/HOSTEDCP-298

Signed-off-by: Gerd Oberlechner <goberlec@redhat.com>

### CODE REVIEW

**Comment by ironcladlou** [comment]
@geoberle Thank you, should be able to take a closer look at this tomorrow. I had to get https://github.com/openshift/hypershift/pull/949 in for some CI work which you'll need to rebase onto. Thanks for your patience!

File: cmd/install/install_render.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+			if !found {
+				return nil, fmt.Errorf("can't resolve patch path %s", k.Path)
+			}
+			nested[k.Field] = k.Value
+		}
+	}
+	return u, nil
+}
+
+func resolveNestedMap(obj map[string]interface{}, path []string) (map[string]interface{}, bool) {
```
[the remark this answers is not in the document]
Comment (geoberle) [reply cmd/install/install_render.go:143]: I will look into it!

File: cmd/install/install_render.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+		return fmt.Errorf("unexpected format %s", format)
+	}
+}
+
+type ObjectPatch struct {
+	Kind      string
+	Name      string
+	Namespace string
+	Path      []string
+	Field     string
```
[the remark this answers is not in the document]
Comment (geoberle) [reply cmd/install/install_render.go:233]: You are right. I will change this

**Comment by geoberle** [comment]
based on what has been implemented in #949 i'm going to change a couple of things in this PR.
i'm setting this to WIP for now

**Review by alvaroaleman — APPROVED** [review]
/tide/merge-method-squash
/hold

Looks good to me, holding so @ironcladlou also gets a chance to have a look

**Comment by ironcladlou** [comment]
/hold cancel

Thanks for all this work! Looking at the diff that was produced in the rendered template, makes me wonder if we should provide an option to write a file per resource so that it's possible to get a sane diff when things change.

**Pull request merged**